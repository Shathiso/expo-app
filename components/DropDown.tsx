import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import SelectDropdown from 'react-native-select-dropdown';
import { FontAwesome } from '@expo/vector-icons';


interface DropDown{
  listData: string[],
  dropDownTitle:string,
  handleSelection: (item:string) => void
}

const DropDown = ({listData, dropDownTitle, handleSelection}: DropDown) => {
  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <Text style={styles.headerTxt}>{dropDownTitle}</Text>
      </View>
      <SelectDropdown
        data={listData}
        onSelect={(selectedItem, index) => {
          handleSelection(selectedItem)
        }}
        renderButton={(selectedItem, isOpen) => {
          return (
            <View style={styles.dropdownButtonStyle}>
              <Text style={styles.dropdownButtonTxtStyle}>
                {(selectedItem) || 'Select'}
              </Text>
              <FontAwesome name={isOpen ? 'chevron-up' : 'chevron-down'} size={18} color="#000" style={styles.dropdownButtonArrowStyle} />
            </View>
          );
        }}
        renderItem={(item, index, isSelected) => {
          return (
            <View
              style={{
                ...styles.dropdownItemStyle,
                ...(isSelected && {backgroundColor: '#D2D9DF'}),
              }}>
              <Text style={styles.dropdownItemTxtStyle}>{item}</Text>
            </View>
          );
        }}
        showsVerticalScrollIndicator={false}
        dropdownStyle={styles.dropdownMenuStyle}
      />
    </View>
  );
};

export default DropDown;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingVertical: 100,
    alignItems: 'center',
    justifyContent: 'space-evenly',
    paddingTop: 3,
    paddingBottom: 3,
    marginTop: 35
  },
  header: {
    position: 'absolute',
    top: 0,
    width: '100%',
    height: 20,
    justifyContent: 'flex-end',
    alignItems: 'flex-start',
    paddingBottom: 16,
    paddingLeft:6
  },
  headerTxt: {
    fontSize: 16,
    paddingBottom:4,
    color: '#151E26',
  },
  dropdownButtonStyle: {
    width: '100%',
    height: 50,
    backgroundColor: '#E9ECEF',
    borderRadius: 12,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    paddingHorizontal: 12,
  },
  dropdownButtonTxtStyle: {
    flex: 1,
    fontSize: 18,
    fontWeight: '500',
    color: '#151E26',
  },
  dropdownButtonArrowStyle: {
    fontSize: 28,
  },
  dropdownButtonIconStyle: {
    fontSize: 28,
    marginRight: 8,
  },
  dropdownMenuStyle: {
    backgroundColor: '#E9ECEF',
    borderRadius: 8,
  },
  dropdownItemStyle: {
    width: '100%',
    flexDirection: 'row',
    paddingHorizontal: 12,
    justifyContent: 'center',
    alignItems: 'center',
    paddingVertical: 8,
  },
  dropdownItemTxtStyle: {
    flex: 1,
    fontSize: 18,
    fontWeight: '500',
    color: '#151E26',
  },
  dropdownItemIconStyle: {
    fontSize: 28,
    marginRight: 8,
  },
});