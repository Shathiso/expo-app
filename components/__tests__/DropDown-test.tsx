import * as React from 'react';
import { render, screen, fireEvent, userEvent } from '@testing-library/react-native';
import DropDown from '../DropDown';


test(`drop down component renders correctly`, () => {
  const handleChange = jest.fn();

  const dropDownList = ['one', 'two', 'three'];

  const tree = render(<DropDown listData={dropDownList} dropDownTitle="name" handleSelection={handleChange}/>).toJSON();

  expect(tree).toMatchSnapshot();;

});
