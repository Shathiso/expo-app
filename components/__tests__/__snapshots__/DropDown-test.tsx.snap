// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`drop down component renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "space-evenly",
      "marginTop": 35,
      "paddingBottom": 3,
      "paddingTop": 3,
      "paddingVertical": 100,
    }
  }
>
  <View
    style={
      {
        "alignItems": "flex-start",
        "height": 20,
        "justifyContent": "flex-end",
        "paddingBottom": 16,
        "paddingLeft": 6,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#151E26",
          "fontSize": 14,
          "paddingBottom": 4,
        }
      }
    >
      name
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#E9ECEF",
        "borderRadius": 12,
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "center",
        "opacity": 1,
        "paddingHorizontal": 12,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#151E26",
          "flex": 1,
          "fontSize": 16,
          "fontWeight": "500",
        }
      }
    >
      Select
    </Text>
    <Text />
  </View>
</View>
`;
